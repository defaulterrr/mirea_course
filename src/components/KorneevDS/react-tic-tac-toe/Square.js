import React from 'react';
import square_styles from './TicTacToe.module.css';

function Square(props){
		return (
			<button className={square_styles.square} onClick={() => props.onClick()}>
			{props.value}
			</button>
		)
}
export default Square;
