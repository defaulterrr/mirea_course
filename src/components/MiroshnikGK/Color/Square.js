import React from 'react';
import Radium from 'radium';
import PropTypes from 'prop-types';

const square = (props) => {
  const { CMYK } = props;
  const { value } = props;
  const h4style = {
    height: '25px',
  };
  const rectKeyframes = Radium.keyframes({
    '0%': { transform: 'rotate(-3deg)' },
    '100%': { transform: 'rotate(3deg)' },
  });
  const rectanglestyle = {
    width: '300px',
    height: '300px',
    background: value,
    marginTop: '30px',
    border: '5px solid white',
    animation: '2s infinite alternate ease-in-out',
    animationName: rectKeyframes,
  };
  return (
    <div className="SquareGK">
      <div className="rectangleGK" style={rectanglestyle} />
      <h4 style={h4style}>
        CMYK:
        <br />
        {CMYK}
      </h4>
    </div>
  );
};

square.propTypes = {
  CMYK: PropTypes.string,
  value: PropTypes.string,
};

square.defaultProps = {
  CMYK: '',
  value: '',
};

export default Radium(square);
