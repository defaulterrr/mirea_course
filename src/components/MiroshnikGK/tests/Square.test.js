import React from 'react';
import { create } from 'react-test-renderer';
import { StyleRoot } from 'radium';
import Square from '../Color/Square';

describe('[RENDER] Square â€“ Snapshot test', () => {
  test('Matches the snapshot', () => {
    const square = create(<StyleRoot><Square /></StyleRoot>);
    expect(square.toJSON()).toMatchSnapshot();
  });
});
