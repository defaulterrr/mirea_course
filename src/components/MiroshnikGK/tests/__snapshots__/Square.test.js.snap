// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Square â€“ Snapshot test Matches the snapshot 1`] = `
<div
  data-radium={true}
>
  <div
    className="SquareGK"
    data-radium={true}
  >
    <div
      className="rectangleGK"
      data-radium={true}
      style={
        Object {
          "animation": "2s infinite alternate ease-in-out",
          "animationName": "radium-animation-baec9553",
          "background": "",
          "border": "5px solid white",
          "height": "300px",
          "marginTop": "30px",
          "width": "300px",
        }
      }
    />
    <h4
      data-radium={true}
      style={
        Object {
          "height": "25px",
        }
      }
    >
      CMYK:
      <br />
      
    </h4>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@keyframes radium-animation-baec9553 {
0%{transform: rotate(-3deg);}
100%{transform: rotate(3deg);}
}
",
      }
    }
  />
</div>
`;
